enhancements/olm/service-account-deprecation.md

```diff
@@ -0,0 +1,262 @@
+---
+title: serviceaccount-field-deprecation
+authors:
+  - "@rashmigottipati"
+reviewers:
+  - "@everettraven"
+  - "@grokspawn"
+  - "@trgeiger"
+  - "@joelanford"
+approvers:
+  - "@everettraven"
+  - "@trgeiger"
+  - "@grokspawn"
+creation-date: 2025-10-08
+last-updated: 2025-10-15
+status: implementable
+---
+
+# Service Account Field Deprecation
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Deprecate the `.spec.serviceAccount` field from the ClusterExtension API in the operator-controller. This field was originally introduced to enforce least privilege by requiring users to provide a ServiceAccount with the necessary RBAC permissions to manage extension content. This proposal removes that requirement and simplifies controller logic and behavior.
+
+**Note**: This deprecation is an interim step. While the intent is to eventually remove the `.spec.serviceAccount` field, we will keep it as optional and mark it as deprecated during this transition period spanning multiple releases. This approach allows users time to adapt without disruption before the field is fully removed.
+
+## Motivation
+
+The original intent of `.spec.serviceAccount` was to support a least-privilege model by allowing users to provide custom `ServiceAccount` with fine-grained permissions. In practice, this design introduced considerable operational and technical complexity, including:
+
+- Token acquisition, rotation, and error handling  
+- Custom `rest.Config` generation and dynamic HTTP clients  
+- Token expiration edge cases 
```

Comment:
Are you actively experiencing issues with being able to handle these scenarios?

From what I remember this logic is all already implemented and is flexed during testing. I'm curious how many times the logic to do these things have been problematic from both an end-user perspective and a maintenance perspective.